api/client/alpn_conn_upgrade.go

```diff
@@ -184,6 +185,8 @@ func newALPNConnUpgradeDialer(dialer ContextDialer, tlsConfig *tls.Config, withP
 		dialer:    dialer,
 		tlsConfig: tlsConfig,
 		withPing:  withPing,
+		// Only use "legacy" mode when it's explicitly set by the env var.
+		useLegacyMode: strings.ToLower(os.Getenv(defaults.TLSRoutingConnUpgradeModeEnvVar)) == "legacy",
```

Comment:
Any reason we need to keep this mode at all?

Reply:
it has better performance than WebSocket since WebSocket introduces another layer of frames, masking etc. just WebSocket has better compatibility. i had a discussion with Edardo once that maybe we should prefer the "legacy" when we can for better performance.

on the other hand, no customer has complained about the upgrade performance nor if i know any customer using `TELEPORT_TLS_ROUTING_CONN_UPGRADE_MODE=legacy`. if we want to deprecate it, i can add warning in this change, and deprecate it in v18?

Reply:
sounds good. i will make a follow up PR to deprecate it, as this PR will be backported. I've updated the changelog stating the mode will be deprecated and i will add the breaking change note in the follow up PR.

Reply:
https://github.com/gravitational/teleport/pull/52573